service/src/main/java/com/commercetools/pspadapter/payone/domain/payone/model/klarna/BaseKlarnaRequest.java

```diff
@@ -0,0 +1,290 @@
+     */
+    public List<String> getId() {
+        return id;
+    }
+
+    public void setId(List<String> id) {
+        this.id = id;
+    }
+
+    /**
+     * Full unit price without discounts.
+     * <p>
+     * (in smallest currency unit! e.g. cent)
+     *
+     * @return list of product prices, respective to {@link #getIt()} and {@link #getId()}
+     */
+    public List<Long> getPr() {
+        return pr;
+    }
+
+    public void setPr(List<Long> pr) {
+        this.pr = pr;
+    }
+
+    /**
+     * @return items quantity, respective to {@link #getId()}
+     */
+    public List<Long> getNo() {
+        return no;
+    }
+
+    public void setNo(List<Long> no) {
+        this.no = no;
+    }
+
+    /**
+     * @return list of items description, respective to {@link #getId()}
+     */
+    public List<String> getDe() {
+        return de;
+    }
+
+    public void setDe(List<String> de) {
+        this.de = de;
+    }
+
+
+    /**
+     * VAT rate (%)
+     * <p>
+     * value < 100 = percent
+     *
+     * @return list of VAT rates in percent, respective to {@link #getId()}
+     */
+    public List<Integer> getVa() {
+        return va;
+    }
+
+    public void setVa(List<Integer> va) {
+        this.va = va;
+    }
+
+    /**
+     * This function will clean/initialize all the list properties(like {@link #id}, {@link #it} and so on) and
+     * populate them from the {@code paymentWithCartLike} (custom)line items, discounts, shipment costs.
+     *
+     * @param paymentWithCartLike instance from which all the prices are fetched.
+     */
+    protected void mapKlarnaPropertiesFromPaymentCart(@Nonnull PaymentWithCartLike paymentWithCartLike) {
+        // create an ordered list of preferred locales to fetch from localized items properties,
+        // like sku, description etc.
+        // The idea is: try first to fetch the locale mentioned in the paymentWithCartLike,
+        // otherwise fallback to the default locale
+        final Locale orderLocale = Locale.forLanguageTag(getPaymentLanguageTagOrFallback(paymentWithCartLike));
+        final List<Locale> localesList = Stream.of(orderLocale, DEFAULT_LOCALE)
+                .distinct() // if orderLocale == DEFAULT_LOCALE
+                .collect(Collectors.toList());
+
+        final CartLike<?> cartLike = paymentWithCartLike.getCartLike();
+        final int itemsCount = cartLike.getLineItems().size()
+                + cartLike.getCustomLineItems().size()
+                + 1 // cartLike.getShippingInfo();
+                + 1; // discount
+
+        this.id = new ArrayList<>(itemsCount);
+        this.it = new ArrayList<>(itemsCount);
+        this.pr = new ArrayList<>(itemsCount);
+        this.no = new ArrayList<>(itemsCount);
+        this.de = new ArrayList<>(itemsCount);
+        this.va = new ArrayList<>(itemsCount);
+
+        // accumulator for full prices - without discounts
+        final AtomicLong accumulatedFullPriceCents = new AtomicLong(0L);
```

Comment:
the reason is different: the closure below (in `forEach` requires  references to be immutable instances, that's why use this _wrapper_: to count inside closures.